editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -133,77 +140,183 @@ void VersionControlEditorPlugin::_initialize_vcs() {
 
 	ERR_FAIL_COND_MSG(!EditorVCSInterface::get_singleton()->initialize(res_dir), "VCS was not initialized");
 
+	EditorVCSInterface::get_singleton()->set_up_credentials(set_up_username->get_text(), set_up_password->get_text());
 	_refresh_stage_area();
+	_refresh_commit_list();
+	_refresh_branch_list();
 }
 
-void VersionControlEditorPlugin::_send_commit_msg() {
+void VersionControlEditorPlugin::_refresh_branch_list() {
+	Array branch_list = EditorVCSInterface::get_singleton()->get_branch_list();
+	branch_select->clear();
+	for (int i = 0; i < branch_list.size(); i++) {
+		branch_select->add_item(branch_list[i], i);
+	}
+
+	// TODO: Add new brach option
+	// branch_select->add_separator();
+	// branch_select->add_item("New Branch");
+	branch_select->select(0); // First branch in the list is current branch
+}
+
+void VersionControlEditorPlugin::_refresh_commit_list() {
 
+	if (!EditorVCSInterface::get_singleton()) {
+		WARN_PRINT("No VCS addon is initialized. Select a Version Control Addon from Project menu");
+		return;
+	}
+
+	commit_list->get_root()->clear_children();
+
+	Array commits_info_list = EditorVCSInterface::get_singleton()->get_previous_commits();
+	for (int i = 0; i < commits_info_list.size(); i++) {
+		Dictionary commit_info = commits_info_list[i];
+		TreeItem *item = commit_list->create_item(commit_list->get_root());
+		item->set_text(0, String(commit_info["message"]).strip_edges());
+		item->set_text(1, String(commit_info["author"]).strip_edges());
+	}
+}
+
+void VersionControlEditorPlugin::_send_commit_msg() {
 	String msg = commit_message->get_text();
 	if (msg == "") {
 
-		commit_status->set_text(TTR("No commit message was provided"));
+		// commit_status->set_text(TTR("No commit message was provided"));
 		return;
 	}
 
-	if (EditorVCSInterface::get_singleton()) {
+	if (!EditorVCSInterface::get_singleton()) {
+		WARN_PRINT("No VCS addon is initialized. Select a Version Control Addon from Project menu");
```

Comment:
Would really like this renamed from addon to plugin because several users raised questions on what this line actually meant